vectorstores/pgvector/engine.go

```diff
@@ -0,0 +1,141 @@
+package pgvector
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/jackc/pgx/v5"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/oauth2/v2"
+	"google.golang.org/api/option"
+)
+
+const Port = 5432 // default PostgreSQL port
+
+type ServiceAccountRetriever interface {
+	GetServiceAccountEmail(ctx context.Context) (string, error)
+}
+type DefaultServiceAccountRetriever struct{}
+
+func (d *DefaultServiceAccountRetriever) GetServiceAccountEmail(ctx context.Context) (string, error) {
+	return getServiceAccountEmail(ctx)
+}
```

Comment:
Should have (at minimal brief) comments on all exported interfaces. 

Separately, this doesn't seem like it really needs an interface?

Reply:
I added the interface to mock the service on the tests, should I use something like mockify instead?